src/components/TextWithEllipsis/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {StyleProp, TextStyle, View, ViewStyle} from 'react-native';
+import Text from '../Text';
+import styles from '../../styles/styles';
+
+type TextWithEllipsisProps = {
+    /** Leading text before the ellipsis */
+    leadingText: string;
+
+    /** Text after the ellipsis */
+    trailingText: string;
+
+    /** Styles for leading and trailing text */
+    textStyle?: TextStyle;
+
+    /** Styles for leading text View */
+    leadingTextParentStyle?: StyleProp<ViewStyle>;
+
+    /** Styles for parent View */
+    wrapperStyle?: StyleProp<ViewStyle>;
+};
+
+function TextWithEllipsis({leadingText, trailingText, textStyle, leadingTextParentStyle, wrapperStyle}: TextWithEllipsisProps) {
+    return (
+        <View style={[styles.flexRow, wrapperStyle]}>
+            <View style={[styles.flexShrink1, leadingTextParentStyle]}>
+                <Text style={textStyle}>{leadingText}</Text>
```

Comment:
```suggestion
                <Text style={textStyle} numberOfLines={1}>{leadingText}</Text>
```

Why did you remove `numberOfLines` prop?

Reply:
Text component was mistyped, that's why probably. @MaciejSWM let's merge the newest main and resolve conflicts (Text is now fixed on main 👍)

Reply:
Yeah. Deleted as it was not a valid prop of Text so I thought it was a mistake to begin with. Restoring,